clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -410,7 +410,21 @@ class ScalarExprEmitter : public StmtVisitor<ScalarExprEmitter, mlir::Value> {
     return Visit(E->getSubExpr());
   }
 
-  mlir::Value VisitUnaryNot(const UnaryOperator *E) { llvm_unreachable("NYI"); }
+  mlir::Value VisitUnaryNot(const UnaryOperator *E) {
```

Comment:
Any specific reason on why not following LLVM's codegen? We usually try to match the skeleton as much as possible. Any reason why this wouldn't work?
```
Value *ScalarExprEmitter::VisitUnaryNot(const UnaryOperator *E) {
  TestAndClearIgnoreResultAssign();
  auto op = Visit(E->getSubExpr());
  // return buildUnaryOp ....
}
```